src/qibolab/transpilers/placer.py

```diff
@@ -0,0 +1,261 @@
+    """Checks if layout is correct.
+
+    Args:
+        layout (dict): physical to logical qubit mapping.
+        verbose (bool): If ``True`` it prints info messages.
+
+    Returns: ``True`` if layout is written in the correct form.
+    otherwise returns ``False``.
+    """
+    values = list(layout.values())
+    values.sort()
+    keys = list(layout.keys())
+    ref_keys = list("q" + str(i) for i in range(len(keys)))
+    if keys != ref_keys:
+        if verbose:
+            log.info("Some physical qubits in the layout may be missing or duplicated")
+        return False
+    if values != list(range(len(values))):
+        if verbose:
+            log.info("Some logical qubits in the layout may be missing or duplicated")
+        return False
+    return True
+
+
+class Trivial(Placer):
+    """Place qubits trivially, same logical and physical placement
+
+    Attributes:
+        connectivity (networkx.Graph): chip connectivity.
+    """
+
+    def __init__(self, connectivity=None):
+        """Args:
+        connectivity (networkx.graph): chip connectivity.
+        """
+        self.connectivity = connectivity
+
+    def __call__(self, circuit: Circuit):
+        """Find the trivial placement for the circuit.
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit model to check.
+        """
+        return dict(zip(list("q" + str(i) for i in range(circuit.nqubits)), range(circuit.nqubits)))
+
+
+class Custom(Placer):
+    """Define a custom initial qubit mapping.
+
+    Attributes:
+        map (list or dict): Physical to logical qubit mapping,
+        example [1,2,0] or {"q0":1, "q1":2, "q2":0} to assign the
+        physical qubits 0;1;2 to the logical qubits 1;2;0 respectively.
+        connectivity (networkx.Graph): chip connectivity.
+    """
+
+    def __init__(self, map, connectivity=None, verbose=False):
+        """Args:
+        map (list or dict): Physical to logical qubit mapping,
+        example [1,2,0] or {"q0":1, "q1":2, "q2":0} to assign the
+        physical qubits 0;1;2 to the logical qubits 1;2;0 respectively.
+        connectivity (networkx.Graph): chip connectivity.
+        verbose (Bool): if "True" print info messages.
+        """
+        self.connectivity = connectivity
+        self.map = map
+        self.verbose = verbose
+
+    def __call__(self, circuit=None):
+        """Return the custom placement if it can be applied to the given circuit (if given).
+
+        Args:
+            circuit (qibo.models.Circuit): Circuit to be transpiled.
+        """
+        if isinstance(self.map, dict):
+            pass
+        elif isinstance(self.map, list):
+            self.map = dict(zip(list("q" + str(i) for i in range(len(self.map))), self.map))
+        else:
+            raise_error(TypeError, "Use dict or list to define mapping.")
+        if circuit is not None:
+            if not check_placement(circuit, self.map, self.verbose):
+                raise_error(ValueError)
```

Comment:
Consider turning `check_placement` into `assert_placement`. 

In this way, you could raise a more informative error inside, and using a custom error type, you can always catch that and only that, and implement `if`s when required:
```py
try:
    assert_placement(...)
    # if branch
except PlacementError:
    # else branch
```

Reply:
ok i will move to assert_placement, for the custom error i don't know what i have to do

Reply:
```py
class PlacementError(ValueError):
    """Error in the placement process."""
```

The docstring is a bit generic. If you can make a more descriptive one, the better.